secure_message/common/utilities.py

```diff
@@ -8,85 +8,67 @@
-    ru_id = None
-    survey = None
-    cc = None
-    label = None
-    desc = True
-    ce = None
-
-    if args.get('limit'):
-        limit = int(args.get('limit'))
-
-    if args.get('page'):
-        page = int(args.get('page'))
-
-    if args.get('ru_id'):
-        string_query_args = add_string_query_args(string_query_args, 'ru_id', args.get('ru_id'))
-        ru_id = str(args.get('ru_id'))
-    if args.get('survey'):
-        survey = str(args.get('survey'))
-        string_query_args = add_string_query_args(string_query_args, 'survey', args.get('survey'))
-    if args.get('cc'):
-        cc = str(args.get('cc'))
-        string_query_args = add_string_query_args(string_query_args, 'cc', args.get('cc'))
+    fields = {'page': 1, 'limit': MESSAGE_QUERY_LIMIT, 'ru_id': None, 'survey': None,
+              'desc': True, 'cc': None, 'label': None, 'ce': None}
+
+    for field in ['survey', 'cc', 'ce', 'ru_id', 'label']:
+        if args.get(field):
+            fields[field] = str(args.get(field))
+
+    for field in ['limit', 'page']:
+        if args.get(field):
+            fields[field] = int(args.get(field))
+
+    if args.get('desc') and args.get('desc') == 'false':
+        fields['desc'] = False
     if draft_only:
-        label = Labels.DRAFT.value
-        string_query_args = add_string_query_args(string_query_args, 'label', label)
-    elif args.get('label'):
-        label = str(args.get('label'))
-        string_query_args = add_string_query_args(string_query_args, 'label', args.get('label'))
-    if args.get('ce'):
-        ce = str(args.get('ce'))
-        string_query_args = add_string_query_args(string_query_args, 'ce', args.get('ce'))
-    if args.get('desc'):
-        desc = False if args.get('desc') == 'false' else True
-        string_query_args = add_string_query_args(string_query_args, 'desc', args.get('desc'))
+        fields['label'] = Labels.DRAFT.value
 
-    return MessageArgs(string_query_args=string_query_args, page=page, limit=limit, ru_id=ru_id, survey=survey,
-                       cc=cc, label=label, desc=desc, ce=ce)
+    return MessageArgs(page=fields['page'], limit=fields['limit'], ru_id=fields['ru_id'],
+                       survey=fields['survey'], cc=fields['cc'], label=fields['label'],
+                       desc=fields['desc'], ce=fields['ce'])
 
 
-def add_string_query_args(string_query_args, arg, val):
-    """Create query string for get messages options"""
-    if string_query_args == '?':
-        return f'{string_query_args}{arg}={val}'
-    return f'{string_query_args}&{arg}={val}'
+def generate_string_query_args(args):
+    string_query = "?"
+    for field in args._fields:
+        if field in ['page']:
+            pass
+        value = getattr(args, field)
+        if value is not None:
+            if string_query == '?':
+                string_query = string_query + f'{field}={value}'
+            else:
+                string_query = string_query + f'&{field}={value}'
```

Comment:
The query string should be generated using `urllib.parse.urlencode` rather than string concatenation (which is error prone and less clear) - see https://docs.python.org/3.6/library/urllib.request.html#urllib-examples 

The `urlencode` method can take a dict or a sequence of two element tuples, so with some changes to how we deal with `MessageArgs` we should be able to remove this method.